### PR TITLE
Try using yarn instead of npm install

### PR DESCRIPTION
**Pull request by pcardune**

### CODE REVIEW

**Comment by wjordan** [comment]
FYI in my own tests I found Yarn had a bug where it won't process the reference `"git://github.com/sedge/grunt-contrib-compress.git#fix-gruntjs#61"` found in our [bramble](https://github.com/code-dot-org/bramble) package. We'll need to [apply this diff](https://github.com/code-dot-org/bramble/compare/cdo-master...wjordan:cdo-master-2) (or change the `@code-dot-org/bramble` reference from `0.1.5` to `"git://github.com/wjordan/bramble#cdo-master-2"`) in order for Yarn to work

**Comment by pcardune** [comment]
@wjordan any reason why that diff has not been merged yet?

**Comment by wjordan** [comment]
no specific reason- general reason is that I only briefly experimented with yarn because I was curious, and didn't [have the time currently to] follow through on it [with more substantial testing of all related use- and edge-cases, taking responsibility for anything that might break afterwards, etc etc].

My initial experiment with yarn was pretty positive though, aside from that one dependency issue (and some intermittent checksum issues). After all of the various issues we've dealt with in npm, I would be a _big_ fan if you do follow through on this!

File: circle.yml [hunk 1]
```diff
@@ -30,12 +30,18 @@ dependencies:
     - google-chrome --version
     - ./bin/circle/upgrade-google-chrome.sh
     - google-chrome --version
+    # Install Yarn
+    - sudo apt-key adv --keyserver pgp.mit.edu --recv D101F7899D41F3C3
+    - echo "deb http://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+    - sudo apt-get update -qq
+    - sudo apt-get install -y -qq yarn
```
Comment (Bjvanminnen) [inline circle.yml:37]: could/should this just be a rake task?

**Comment by Bjvanminnen** [comment]
https://github.com/code-dot-org/code-dot-org/blob/staging/apps/Gruntfile.js#L635 Should this be updated?

**Comment by Bjvanminnen** [comment]
Should also update the README where we talk about doing npm install before merging.

File: circle.yml (hunk 1 above)
Comment (pcardune) [reply circle.yml:37]: maybe? what does a rake task buy us?

**Comment by pcardune** [comment]
that firebase installer thing is kind of weird. why is that whole grunt task even necessary?

**Comment by Bjvanminnen** [comment]
Talk to Dave I think about firebase installer

File: circle.yml (hunk 1 above)
Comment (Bjvanminnen) [reply circle.yml:37]: Seems reasonable to have these steps run as part of rake install actually. Means fewer steps to get setup. Also means we dont need to worry about having circle steps get out of sync with our readme instructions.

**Comment by wjordan** [comment]
firebase grunt task PR is here: #9098.

File: circle.yml (hunk 1 above)
Comment (pcardune) [reply circle.yml:37]: I agree with that. Unfortunately, this setup is OS dependent.

**Comment by pcardune** [comment]
Ok, I made some changes and looks like tests are passing still. @Bjvanminnen and @wjordan, could you guys PTAL?

File: SETUP.md [hunk 2]
```diff
@@ -38,32 +38,34 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
...
   1. `npm --version   # --> 3.10.8`
+  1. `yarn -V         # --> 0.16.1`
 
 ### Ubuntu 14.04
 
 1. `sudo apt-get update`
 1. `sudo apt-get install -y git mysql-server mysql-client libmysqlclient-dev libxslt1-dev libssl-dev zlib1g-dev imagemagick libmagickcore-dev libmagickwand-dev openjdk-7-jre-headless libcairo2-dev libjpeg8-dev libpango1.0-dev libgif-dev curl pdftk enscript libsqlite3-dev phantomjs build-essential`
   * **Hit enter and select default options for any configuration popups, leaving mysql passwords blank**
-1. Upgrade npm to 2.0. If `npm -v` says less than 2.0 then
+1. Upgrade npm to 3.10.8 If `npm -v` says less than 3.10.8 then
```
Comment (Bjvanminnen) [inline SETUP.md:68]: Does npm version matter if we're using yarn?

Comment (pcardune) [reply SETUP.md:68]: It probably doesn't matter. I just noticed that the npm versions we were talking about in this README were inconsistent between ubuntu and mac instructions, so I updated it. Also it seems like the easiest way to install yarn is with npm (which I find rather amusing), so maybe having a later npm version is a good idea?

**Comment by pcardune** [comment]
I'm going to rebase this onto staging and see if we are still green. If we ware, how do people feel about merging this tomorrow after the MC stuff goes out?

**Comment by Bjvanminnen** [comment]
Merging tomorrow seems fine to me.

**Pull request merged**